### ISSUE TITLE
Database Optimization causing up/down time on sites

### ISSUE THREAD

**Issue by curtismchale** [proudcity/wp-proudcity#2327]
We've been having some SQL queries causing up/down time issues on sites. This ticket is here to track the work for it.

**Comment by curtismchale** [comment]
This is a fairly complete account of what I know so far about our databases @lukefretwell @kevindherman 

## Broken Link Checker

While I did run the Broken Link Checker "nuclear" option on Hellam (it does all it's indexing again) it did **not** take the site down. This isn't conclusive evidence that BLC is not part of the problem, it just didn't cause another issue when I tried.

I did find issues with BLC database entries in ELGL that were surfaced in our SQL query views that we look at when sites are slow. Specifically even though BLC was inactive on the site there were big queries running against the tables found and the entries for BLC in those tables were about 1/3 of the database size of ELGL, totalling around 300mb.

On ELGL I removed the BLC plugin (it will come back inactive on pod restart currently) and then manually removed all the BLC tables on the site so there is nothing to run any type of query against. This cut the database by 1/3 and since there were no tables in ELGL for BLC the long queries being issued against the tables stopped.

At times I've seen similar issues with the BLC tables for both `proudcity.com` and `holyoke.org` though since the online tool to visualize big queries are time based, they don't always show the BLC tables being an issue as it's scanning of the sites is only done periodically.

Overall, I think we need to move away from BLC, but we already have an issue for that: [#2298](https://github.com/proudcity/wp-proudcity/issues/2298)

## `wp_options` entries

We also have some odd queries running against `wp_options`. `wp_options` is the main place in WordPress that plugins store settings. I'm waiting on Jeff to give me some more permissions in our `gcloud` instance so I can run full traces on the queries being run. [This is the relevant Slack thread where I'm waiting](https://proudcity.slack.com/archives/C0AH683JQ/p1691600498906179).

Some sites (San Rafael at one point today) are also running an odd `wp_options` query that looks like this: 

```sql
SELECT `option_value` FROM `wp_options` WHERE `option_name` = ? LIMIT ?
```

I need the extra permissions to run a full trace on it, but the extra odd thing is that San Rafael is running this 1,000,000 times an hour. There are a few other sites running this just under 1,000,00 times an hour, then there is a steep drop off in how often this is run. From talking to some other devs, they have seen similar issues when `wp_options` gets too large. Specifically they've seen different caching layers have issues because the caching layer has a 1mb limit per cached item, and some of our `wp_options` items are far larger than that. As of 15:27 today there are 4 sites that have called this query over 1,000,000 times in the last 6 hours.

One such item is the database entry for Admin Menu Editor which is almost 300mb on San Rafael, and since the menus are largely the same it will be about the same on other sites. I did run the "compress" option on the plugin settings for Admin Menu Editor for San Rafael to see if this makes any difference in the number of times that the query listed above runs.

Once I get the permissions to see a full trace on the queries we should be able to track the exact query down. We already have an issue listed for hardcoding the admin menu order: [#2315](https://github.com/proudcity/wp-proudcity/issues/2315)
After checking a handful of sites these are other large entries in `wp_options`. They don't show up on every site, but they're behind the `ws_menu_editor` entry in top 10 lists of huge options entries on sites. I am skipping things from WordPress SEO because we use/need that plugin currently.

- `jetpack_file_data` - we probably don't need this since we don't install jetpack
- Gravity Flow - it has a long seemingly random string as it's name so I need to dig deeper to see what's up on other sites to see if it has the same name and we can automate removal
- `wsblc_options` - Broken Link Checker settings
- `blc_installation_log` - Broken Link Checker
- `monster_insights_report_data_overview` - on `proudcity.com` so we must have used a Monster Insights plugin at one point

## Manual Database Clean Up

As I've checked up on individual sites that show up in one of the various views for big queries, or queries that are running all the time, I've found many sites with random tables from plugins we don't use anymore. Like Business Directory Plugin (see [Slack](https://proudcity.slack.com/archives/C0AH683JQ/p1691533753057189)) which was on ELGL.

Gravity Flow tables are present on some sites, though we don't use it anymore (see [Slack](https://proudcity.slack.com/archives/C0AH683JQ/p1691601736555159)).

Events Organizer tables are on some sites and I don't think we use that.
- `wp_eo_events`, `wp_eo_venuemeta`

I'm sure there are more. These table stick around because either the plugin devs didn't build a "clean up" routine in the plugin to delete the extra data or, it never got run because we just removed the plugins from our standard setup and then they didn't get installed which effectively deactivates them on sites because they're no longer present.

I hate to say it, but we may need to look at each site database to find odd tables that are no longer needed and then manually remove them.

## Outstanding database tickets

We also have 2 outstanding issues for long queries that I haven't had a chance to look deeper at because I don't have the full trace permissions.
- [#2221](https://github.com/proudcity/wp-proudcity/issues/2221)
- [#2220](https://github.com/proudcity/wp-proudcity/issues/2220)

Both of these types of queries continue to be present on various views when I look at what is being called lots, or taking a long time on our databases.

## More To Do

I still need to investigate Gravity Flow to see if it has the same names on all sites and can be deleted. I need to trace the queries to see where they are coming from in a number of instances.

The next thing I need from Jeff is the permissions to trace queries fully.

**Comment by lukefretwell** [comment]
@curtismchale 

Agree:

> jetpack_file_data - we probably don't need this since we don't install jetpack

We tested this prior to deciding on Sitekit:

> monster_insights_report_data_overview - on proudcity.com so we must have used a Monster Insights plugin at one point
